integration-tests/testsetups/keeper_benchmark.go

```diff
@@ -0,0 +1,348 @@
+			}
+		}
+	}
+
+	for index := range inputs.RegistryVersions {
+		log.Info().Int("Index", index).Msg("Starting Test Setup")
+
+		linkToken, err := contractDeployer.DeployLinkTokenContract()
+		Expect(err).ShouldNot(HaveOccurred(), "Deploying Link Token Contract shouldn't fail")
+		err = k.chainClient.WaitForEvents()
+		Expect(err).ShouldNot(HaveOccurred(), "Failed waiting for LINK Contract deployment")
+
+		k.keeperRegistries[index], k.keeperConsumerContracts[index], k.upkeepIDs[index] = actions.DeployBenchmarkKeeperContracts(
+			inputs.RegistryVersions[index],
+			inputs.NumberOfContracts,
+			uint32(inputs.UpkeepGasLimit), //upkeepGasLimit
+			linkToken,
+			contractDeployer,
+			k.chainClient,
+			k.Inputs.KeeperRegistrySettings,
+			inputs.BlockRange,
+			inputs.BlockInterval,
+			inputs.CheckGasToBurn,
+			inputs.PerformGasToBurn,
+			inputs.FirstEligibleBuffer,
+			inputs.PreDeployedConsumers,
+			inputs.ResetUpkeeps,
+			inputs.UpkeepResetterAddress,
+		)
+	}
+
+	for index := range inputs.RegistryVersions {
+		// Fund chainlink nodes
+		err = actions.FundChainlinkNodesAddress(k.chainlinkNodes, k.chainClient, k.Inputs.ChainlinkNodeFunding, index)
+		Expect(err).ShouldNot(HaveOccurred(), "Funding Chainlink nodes shouldn't fail")
+	}
+
+	log.Info().Str("Setup Time", time.Since(startTime).String()).Msg("Finished Keeper Benchmark Test Setup")
+	err = k.SendSlackNotification(nil)
+	if err != nil {
+		log.Warn().Msg("Sending test start slack notification failed")
+	}
+}
+
+// Run runs the keeper benchmark test
+func (k *KeeperBenchmarkTest) Run() {
+	k.TestReporter.Summary.Load.TotalCheckGasPerBlock = int64(k.Inputs.NumberOfContracts) * k.Inputs.CheckGasToBurn
+	k.TestReporter.Summary.Load.TotalPerformGasPerBlock = int64((float64(k.Inputs.NumberOfContracts) / float64(k.Inputs.BlockInterval)) * float64(k.Inputs.PerformGasToBurn))
+	k.TestReporter.Summary.Load.AverageExpectedPerformsPerBlock = float64(k.Inputs.NumberOfContracts) / float64(k.Inputs.BlockInterval)
+	k.TestReporter.Summary.TestInputs = map[string]interface{}{
+		"NumberOfContracts":   k.Inputs.NumberOfContracts,
+		"BlockCountPerTurn":   k.Inputs.KeeperRegistrySettings.BlockCountPerTurn,
+		"CheckGasLimit":       k.Inputs.KeeperRegistrySettings.CheckGasLimit,
+		"MaxPerformGas":       k.Inputs.KeeperRegistrySettings.MaxPerformGas,
+		"CheckGasToBurn":      k.Inputs.CheckGasToBurn,
+		"PerformGasToBurn":    k.Inputs.PerformGasToBurn,
+		"BlockRange":          k.Inputs.BlockRange,
+		"BlockInterval":       k.Inputs.BlockInterval,
+		"UpkeepSLA":           k.Inputs.UpkeepSLA,
+		"FirstEligibleBuffer": k.Inputs.FirstEligibleBuffer,
+		"NumberOfRegistries":  len(k.keeperRegistries),
+	}
+	startTime := time.Now()
+	k.TestReporter.Summary.StartTime = startTime.UnixMilli() - (90 * time.Second.Milliseconds())
+
+	rampUpBlocks := int64(k.Inputs.NumberOfContracts) / int64(k.TestReporter.Summary.Load.AverageExpectedPerformsPerBlock*2)
```

Comment:
as discussed offline, this can be removed in favour of doing a batch reset just before the test starts

Reply:
have both batch reset and on-the-fly reset options available, configurable via a test input.